src/Agent.Listener/CommandSettings.cs

```diff
@@ -127,6 +127,39 @@ public bool GetRunAsService()
                 defaultValue: false);
         }
 
+        public bool GetEnableAutoLogon()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.EnableAutoLogon,
+                description: StringUtil.Loc("EnableAutoLogon"),
+                defaultValue: !Unattended);
+        }
+
+        public bool GetRestartPermission()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.RestartIfNeeded,
+                description: StringUtil.Loc("RestartNow"),
```

Comment:
There is a case for skipping the restart by the user and he does it later.

---
In reply to: [115208674](https://github.com/Microsoft/vsts-agent/pull/951#discussion_r115208674) [](ancestors = 115208674)